### DISCUSSION TITLE
SX1262: How to see the actual duty cycle from startReceiveDutyCycleAuto()

### DISCUSSION THREAD

**Discussion by MrSniffer** [jgromes/RadioLib#585] [Q&A]
I want to learn what RXperiod and RXsleep periods were selected when I use startReceiveDutyCycleAuto(). I'm doing a power budget calculation and I don't have equipment to look at the real world current draw cycling. For example, in my case I have BW 500.0, SF 9, CR 4/6, Preamble length 32. There is a TCXO and DC-DC converter.

**Comment by jgromes** [comment]
If you need to know the sleep and Rx(/wake) periods that were calculated, it's in the debug (non verbose) output. The values should be microseconds. Keep in mind that since you're using TCXO, there will be an extra delay between sleep and wake (5000 us by default).

**Reply by MrSniffer** [reply]
Indeed there is 5000us for the xtal sync (3.8mA), and then the RX period (8.0mA) followed by sleep(~uA). Below is a typical pattern, FYI. But I am surprised that it appears to me that when I call startReceiveDutyCycleAuto(0,8); the 5000 us period begins right away, but the API doesn't return until after the initial 5000us. I looked at the SX126x.cpp and it appears to me that this may just be the way the chip is implemented. Not a tragedy, but I can't turn off my MCU etc. until after the call returns, e.g. wasting power for 5ms. Doesn't seem like anything to do here, just a note for the future.

![CubeCell DutyCycle-2](https://user-images.githubusercontent.com/9292529/194810029-1a3eaf7d-6d4b-4694-ab1e-e42b642d689b.PNG)
